ciao/arch/architecture.md

```diff
@@ -0,0 +1,423 @@
+---
+title: The Architecture of the Cloud Integrated Advanced Orchestrator (CIAO)
+permalink: architecture.html
+keywords: CIAO, architecture
+---
+
+## A whirlwind tour of the Cloud Integrated Advanced Orchestrator
+
+The Cloud Integrated Advanced Orchestrator (CIAO) is designed to be a simple
+IAAS solution for creating private clouds within an enterprise.  A CIAO cluster
+is composed of a number of different computing devices referred to as nodes.  It
+can be used to deploy units of computation, known as instances, across its
+nodes.  Once deployed, instances can be monitored and managed.  Two types of
+instances are supported; VM based instances, in which all computation is
+performed inside a hypervisor, and container instances, in which docker
+containers are run directly on the cluster nodes.
+
+Instances are created from workloads and images.  A workload is a set of
+instructions for creating instances of a particular type.  They, for example,
+specify the type of instance to be created, i.e., VM or container, its CPU,
+memory and storage requirements, its backing image and a set of instructions to
+run inside the instance when it is started.  Multiple instances can be created
+from the same workload.  Each instance has its own disk image that contains the
+instance's rootfs.  This disk image is derived from the backing image specified
+in the workload from which it was created.  For VM instances, both the instance
+image and the backing image are stored in a [ceph](http://docs.ceph.com)
+cluster.  For container images, the backing image refers to a docker container
+image supplied by a docker registry.  Both the container backing image and the
+container rootfs image are stored locally on the compute node on which the
+container runs.  Additional data volumes can be created in the ceph cluster and
+attached to both container and VM instances at either creation time or at a
+later date.
```

Comment:
I thought we cannot add volumes to a running container?

Reply:
You can't but you can add volumes to a stopped container.